bundles/binding/org.openhab.binding.lcn/src/main/java/org/openhab/binding/lcn/LcnBindingProvider.java

```diff
@@ -0,0 +1,194 @@
+	
+	/**
+	 * Stores all item names related to an LCN module address.
+	 * Used for optimization.
+	 * This map is never cleaned and might contain out-dated mappings or items that no longer exist.
+	 */ 
+	private final HashMap<LcnAddrMod, HashSet<String>> itemNamesByModulCache = new HashMap<LcnAddrMod, HashSet<String>>();
+	
+	/**
+	 * Item processing for the LCN bindings.
+	 * {@inheritDoc}
+	 */
+	@Override
+	public void processBindingConfiguration(String context, Item item, String bindingConfig) throws BindingConfigParseException {
+		super.processBindingConfiguration(context, item, bindingConfig);
+		Matcher matcher = PATTERN_BINDING_GENERAL.matcher(bindingConfig);
+		if (!matcher.matches())
+			throw new BindingConfigParseException(bindingConfig + "' contains no valid binding!");
+		matcher.reset();
+		LcnBindingItem bc = new LcnBindingItem(item);		
+		while (matcher.find()) {
+			String binding = matcher.group(1);
+			if (binding != null && !binding.trim().isEmpty()) {
+				String openHabCmd = null;
+				String connId, lcnTarget;
+				Matcher openHabMatcher = PATTERN_BINDING_WITH_OPENHAB.matcher(binding);
+				Matcher pureMatcher = PATTERN_BINDING_PURE.matcher(binding);
+				if (openHabMatcher.matches()) {
+					openHabCmd = openHabMatcher.group(1);
+					connId = openHabMatcher.group(2);
+					lcnTarget = openHabMatcher.group(3);	
+				} else if (pureMatcher.matches()) {
+					connId = pureMatcher.group(1);
+					lcnTarget = pureMatcher.group(2);
+				} else
+					throw new BindingConfigParseException("Invalid binding configuration for " + binding + "!");
+				String lcnShort = resolveMappings(lcnTarget, openHabCmd);
+				if (lcnShort == null || lcnShort.equals(openHabCmd))
+					lcnShort = lcnTarget;
+				Command cmd = openHabCmd == null ?
+					TypeParser.parseCommand(new StringItem("").getAcceptedCommandTypes(), "") :
+					openHabCmd.equals("%i") ? new StringType("%i") :
+					TypeParser.parseCommand(item.getAcceptedCommandTypes(), openHabCmd);
+				bc.add(new LcnBindingItem.Binding(cmd, connId, lcnShort));
+			}
+		}
+		// Finished
+		this.addBindingConfig(item, bc);
+		for (LcnAddrMod addr : bc.getRelatedModules()) {
+			HashSet<String> l = this.itemNamesByModulCache.get(addr);
+			if (l == null) {
+				l = new HashSet<String>();
+				this.itemNamesByModulCache.put(addr, l);
+			}
+			l.add(item.getName());
+		}
+	}
+	
+	/** {inheritDoc} */
+	@Override
+	public String getBindingType() {
+		return "lcn";
+	}
+
+	/** {inheritDoc} */
+	@Override
+	public void validateItemType(Item item, String bindingConfig) throws BindingConfigParseException {
+		// Everything is accepted
+	}
+	
+	/**
+	 * Resolves LCN commands (with mappings) to plain commands.
+	 * 
+	 * @param lcnTarget the target or a mapping
+	 * @param openHABcmd the command send by openHAB
+	 * @return the resolved result (can be null)
+	 */
+	private static String resolveMappings(String lcnTarget, String openHABcmd) {
```

Comment:
the main purpose of this method seem to be getting and executing the TransformationService. i would tend to rename it …